src/github.com/matrix-org/go-neb/api.go

```diff
@@ -202,8 +203,29 @@ func (s *configureClientHandler) OnIncomingRequest(req *http.Request) (interface
 }
 
 type configureServiceHandler struct {
-	db      *database.ServiceDB
-	clients *clients.Clients
+	db               *database.ServiceDB
+	clients          *clients.Clients
+	mapMutex         sync.Mutex
+	mutexByServiceID map[string]*sync.Mutex
+}
+
+func newConfigureServiceHandler(db *database.ServiceDB, clients *clients.Clients) *configureServiceHandler {
+	return &configureServiceHandler{
+		db:               db,
+		clients:          clients,
+		mutexByServiceID: make(map[string]*sync.Mutex),
+	}
+}
+
+func (s *configureServiceHandler) getMutexForServiceID(serviceID string) *sync.Mutex {
+	s.mapMutex.Lock()
+	defer s.mapMutex.Unlock()
+	m := s.mutexByServiceID[serviceID]
+	if m == nil {
+		m = &sync.Mutex{}
+		s.mutexByServiceID[serviceID] = m
```

Comment:
I'd prefer that you did something sensible like a striped mutex, (or just used a single mutex for all services here.). I'm not really happy with the idea of creating N mutexes where N == number of services.